ament_tools/build_types/ament_cmake.py

```diff
@@ -153,11 +153,21 @@ def on_test(self, context):
         assert context.build_tests
         # Figure out if there is a setup file to source
         prefix = self._get_command_prefix('test', context)
-        if has_make_target(context.build_space, 'test') or context.dry_run:
-            yield BuildAction(prefix + [MAKE_EXECUTABLE, 'test'])
+        if not IS_WINDOWS:
+            if has_make_target(context.build_space, 'test') or context.dry_run:
+                yield BuildAction(prefix + [MAKE_EXECUTABLE, 'test'])
+            else:
+                self.warn("Could not run test for ament_cmake package because it "
+                          "has no 'test' target")
         else:
-            self.warn("Could not run test for ament_cmake package because it "
-                      "has no 'test' target")
+            if MSBUILD_EXECUTABLE is None:
+                raise VerbExecutionError("Could not find 'msbuild' executable")
+            run_tests_project_file = project_file_exists_at(context.build_space, 'RUN_TESTS')
+            if run_tests_project_file is not None or context.dry_run:
+                yield BuildAction(prefix + [MSBUILD_EXECUTABLE, run_tests_project_file])
+            else:
+                self.warn("Could not locate RUN_TESTS VS project file, "
+                          "did you build with the --build-tests option?")
```

Comment:
I'm not sure that assert will always work, since the user might do this:

```
$ ament build
$ ament test --build-tests
```

On the other hand they might:

```
$ ament build --build-tests
$ ament test
```

Does the value in context persist between runs of different verbs?